### ISSUE TITLE
"Sync is paused"

### ISSUE THREAD

**Issue by felipehw** [flathub/com.google.Chrome#14]
Every time I re-open the browser, my google acount sync is "Paused".
I see this message: *"Sync is paused"*.

This not happen with RPM Chromium.

- My flatpak version: `Flatpak 1.8.2`
- About the Chrome Flatpak:

```
$ flatpak info com.google.Chrome 

Google Chrome - The web browser from Google

          ID: com.google.Chrome
         Ref: app/com.google.Chrome/x86_64/beta
        Arch: x86_64
      Branch: beta
     Version: 85.0.4183.83
     License: LicenseRef-proprietary
      Origin: flathub-beta
  Collection: org.flathub.Beta
Installation: system
   Installed: 12,5 MB
     Runtime: org.freedesktop.Platform/x86_64/19.08
         Sdk: org.freedesktop.Sdk/x86_64/19.08

      Commit: 45520a18ba3b942a0ac9508900c5e64907033d0e4c603f02ab435d7c9db7c235
      Parent: 6d681ceefda3ecf4352c6fb27a060783a3e663ec20b51408a9f47cf886e8610b
     Subject: Update Chrome to 85.0.4183.102 (0a2ccf74)
        Date: 2020-09-08 21:24:44 +0000
```

**Comment by damianatorrpm** [comment]
affects me as well

**Comment by refi64** [comment]
Anything interesting if you run from the terminal? Could be related to #9.

**Comment by felipehw** [comment]
@refi64 

```
$ com.google.Chrome 
[2 preload-host-spawn-strategy] Running: /app/bin/zypak-helper child - /app/extra/chrome --type=zygote
LaunchProcess: failed to execvp:
/app/extra/nacl_helper
[15:15:0919/132118.881799:ERROR:nacl_fork_delegate_linux.cc(322)] Bad NaCl helper startup ack (0 bytes)
Stub sandbox ignoring command: /app/extra/nacl_helper
[1:2:0919/132118.952482:ERROR:nacl_fork_delegate_linux.cc(322)] Bad NaCl helper startup ack (0 bytes)
Gtk-Message: 13:21:19.308: Failed to load module "canberra-gtk-module"
Gtk-Message: 13:21:19.308: Failed to load module "canberra-gtk-module"
[2:33:0919/132119.316689:ERROR:bus.cc(393)] Failed to connect to the bus: Failed to connect to socket /run/dbus/system_bus_socket: No such file or directory
[2:28:0919/132119.375720:ERROR:address_tracker_linux.cc(214)] Could not bind NETLINK socket: Address already in use (98)
[45:45:0919/132119.441623:ERROR:sandbox_linux.cc(374)] InitializeSandbox() called with multiple threads in process gpu-process.
[2:65:0919/132119.551443:ERROR:token_service_table.cc(140)] Failed to decrypt token for service AccountId-115988613736401780345
[2:31:0919/132119.601181:ERROR:nacl_browser.cc(308)] Failed to open NaCl IRT file "/app/extra/nacl_irt_x86_64.nexe": -4
[2:124:0919/132119.698247:ERROR:bus.cc(393)] Failed to connect to the bus: Failed to connect to socket /run/dbus/system_bus_socket: No such file or directory
[2:124:0919/132119.698310:ERROR:bus.cc(393)] Failed to connect to the bus: Failed to connect to socket /run/dbus/system_bus_socket: No such file or directory
[2:124:0919/132119.698383:ERROR:bus.cc(393)] Failed to connect to the bus: Failed to connect to socket /run/dbus/system_bus_socket: No such file or directory
[2:124:0919/132119.698433:ERROR:bus.cc(393)] Failed to connect to the bus: Failed to connect to socket /run/dbus/system_bus_socket: No such file or directory
[2:124:0919/132119.698491:ERROR:bus.cc(393)] Failed to connect to the bus: Failed to connect to socket /run/dbus/system_bus_socket: No such file or directory
[2:2:0919/132119.699406:ERROR:account_info_fetcher.cc(60)] OnGetTokenFailure: Invalid credentials (credentials missing).
[2:77:0919/132120.370545:ERROR:gcm_store_impl.cc(941)] Failed to restore security token.
[2:31:0919/132121.164518:ERROR:registration_request.cc(263)] Registration response error message: PHONE_REGISTRATION_ERROR
[2:31:0919/132121.171663:ERROR:registration_request.cc(263)] Registration response error message: PHONE_REGISTRATION_ERROR
[2:31:0919/132121.173583:ERROR:registration_request.cc(263)] Registration response error message: PHONE_REGISTRATION_ERROR
[2:31:0919/132121.322074:ERROR:mcs_client.cc(706)]   Error code: 500  Error message: Cannot parse data.
[2:31:0919/132121.322098:ERROR:mcs_client.cc(708)] Failed to log in to GCM, resetting connection.
[45:45:0919/132133.258212:ERROR:gl_surface_presentation_helper.cc(259)] GetVSyncParametersIfAvailable() failed for 1 times!

```

**Comment by kaimast** [comment]
I have the same error message with regards to GCM.

**Comment by refi64** [comment]
still trying to figure this out, by any chance, did any of you have a previous proprietary Chrome (i.e. the official upstream binaries, not any distro or other Chromium) installation on the *host* system?

**Comment by kaimast** [comment]
I didn't. I actually installed chrome on a new laptop.

Let me know if you would want me to debug more. (I have a regular Chrome install on my machine now though)

**Comment by iqovoid** [comment]
I also have this problem, I installed fresh fedora silverblue, and the issue occurs on it.

**Comment by e-user** [comment]
Same problem here, on Fedora Silverblue 32, Flatpak 1.8.2.

**Comment by refi64** [comment]
As a quick FYI, I haven't forgotten and am still looking into this, it's
just really really hard to debug. I've also run into it, weirdly enough,
from xdg-open, where opening a URL sometimes causes a new instance to open
with sync paused that never loads pages.

**Comment by e-user** [comment]
@refi64 because of the errors of Chrome not being able to connect to the system bus, as in

```Failed to connect to the bus: Failed to connect to socket /run/dbus/system_bus_socket: No such file or directory```

I ran a custom Flatpak build with `--socket=system-bus` set, and voila, Chrome restarts don't cause sync to stop, or loss of cookies on other websites or extensions. Providing access to the system bus seems to be discouraged according to https://docs.flatpak.org/en/latest/sandbox-permissions.html, so I did some digging about Chrome and accessing D-Bus. I came across a comment at https://github.com/systemd/systemd/issues/11293 which might be related:

```
Also try to use Chromium (or Chrome, I guess) when DBUS_SESSION_BUS_ADDRESS is unset; all cookies should be inaccessible because the "Safe Storage" secret cannot be fetched from the keyring.
```

I checked the running Chrome instance using `flatpak enter` and it shows that DBUS_SESSION_BUS_ADDRESS is correctly set; so the question remains why Chrome insists on connecting to the system bus instead of the session bus.

However, I'm still getting

```
[2:47:1010/134513.249227:ERROR:token_service_table.cc(140)] Failed to decrypt token for service AccountId-100727953024930405246
[2:2:1010/134515.956434:ERROR:password_sync_bridge.cc(362)] Passwords datatype error was encountered: Failed to load entries from password store. Encryption service failure.
```

and attempting to store passwords consequently doesn't work, so #9 is probably an unrelated problem.

**Comment by refi64** [comment]
I don't think the systemd issue is entirely related, since secrets are stored over D-Bus so without D-Bus they wouldn't be able to be stored...

**Comment by refi64** [comment]
https://github.com/flathub/com.google.Chrome/commit/6758c026a6a0380cad72c8253415f2a51c6fa9e7 should fix manually restarting Chrome, in terms of "I close it and re-open but it no longer syncs", I suspect it's related to an upstream bug when udev is not working that manifested itself in the Chromium Flatpak, basically it stays running after you try to close it so it keeps the locks on the session data.

**Comment by fedelibre** [comment]
> I don't think the systemd issue is entirely related, since secrets are stored over D-Bus so without D-Bus they wouldn't be able to be stored...

So `org.freedesktop.secrets=talk` should be enough for storing passwords locally, right?
These are default permissions:

```
$ flatpak info --show-permissions com.google.Chrome 
[Context]
shared=network;ipc;
sockets=x11;wayland;pulseaudio;cups;
devices=all;
filesystems=host-etc;xdg-run/pipewire-0;xdg-music;xdg-documents;xdg-videos;xdg-download;

[Session Bus Policy]
org.gnome.SessionManager=talk
org.freedesktop.Notifications=talk
org.mpris.MediaPlayer2.chrome.*=own
org.freedesktop.secrets=talk
org.freedesktop.FileManager1=talk

[System Bus Policy]
org.freedesktop.Avahi=talk
org.freedesktop.UPower=talk

[Environment]
GTK_PATH=/app/lib/gtkmodules
```

But if passwords are saved on Google account and cannot be used, then it's a sync issue.

I'm currently affected by this issue. Every time I restart Chrome, the Google account appear in pause and I have to log in again and pass the 2 factor authentication (annoying).

But after I'm logged in and the sync appear active, I still cannot access the passwords (stored on Google account).

**Comment by fedelibre** [comment]
> I ran a custom Flatpak build with `--socket=system-bus` set, and voila, Chrome restarts don't cause sync to stop, or loss of cookies on other websites or extensions.

Adding that permission did not help me. I'm using the flatpak from flathub-beta.

Can you show your custom Flatpak build?

**Comment by mmdmine** [comment]
For me affects both flatpak package and aur package.

**Comment by rayfoss** [comment]
Having the issue on Chrome Stable 88 in a new Disco Donkey 64bit install on a Raspberry Pi. Strangely, seahorse is working but the Chromium entry showed missing characters.

**Comment by refi64** [comment]
Fixed by #40.
